pyproject.toml

```diff
@@ -1,14 +1,116 @@
+[project]
+name = 'asdf_astropy'
+description = 'ASDF serialization support for astropy'
+readme = 'README.rst'
+requires-python = '>=3.8'
+license = { file = 'LICENSE.rst' }
+authors = [{ name = 'The Astropy Developers', email = 'astropy.team@gmail.com' }]
+classifiers = [
+    'Programming Language :: Python',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+    'Development Status :: 5 - Production/Stable',
+]
+dependencies = [
+    'astropy >=5.0.4',
+    'asdf >=2.8.0',
+    'numpy',
+    'asdf-coordinates-schemas',
+    'asdf-transform-schemas >=0.2.2',
+    'importlib_resources >=3; python_version <"3.9"',
+    'packaging >=16.0',
+]
+dynamic = ['version']
+
+[project.optional-dependencies]
+test = [
+    'pytest-astropy',
+    'coverage',
+    'scipy',
+]
+docs = [
+    'sphinx',
+    'sphinx-astropy',
+    'sphinx-automodapi',
+    'sphinx-asdf',
+    'graphviz',
+    'matplotlib',
+    'docutils',
+]
+
+[project.urls]
+'tracker' = 'https://github.com/astropy/asdf-astropy/issues'
+'documentation' = 'https://asdf-astropy.readthedocs.io/en/latest/'
+'repository' = 'https://github.com/astropy/asdf-astropy'
+
+[project.entry-points]
+'asdf.extensions' = { asdf-astropy = 'asdf_astropy.integration:get_extensions' }
+'asdf.resource_mappings' = { asdf-astropy = 'asdf_astropy.integration:get_resource_mappings' }
+
 [build-system]
-requires = ["setuptools>=42",
-            "setuptools_scm[toml]>=3.4",
-            "wheel",
-            "oldest-supported-numpy",
-	    ]
+requires = [
+    "setuptools>=42",
```

Comment:
I think this should be `setuptools>=60` because that's the version that "officially" supports `pyproject.toml`. I also need to go through the other repo PRs and update that